src/applications/disability-benefits/wizard/pages/bdd.jsx

```diff
@@ -35,15 +36,17 @@ const findNextPage = state => {
   const differenceBetweenDatesInDays =
     dateDischarge.diff(dateToday, 'days') + 1;
 
-  if (differenceBetweenDatesInDays < 90) {
-    saveDischargeDate(dateDischarge, false);
-    return pageNames.fileClaimEarly;
-  } else if (differenceBetweenDatesInDays <= 180) {
-    saveDischargeDate(dateDischarge, true);
-    return pageNames.fileBDD;
+  if (differenceBetweenDatesInDays > 0) {
+    if (differenceBetweenDatesInDays < 90) {
+      saveDischargeDate(dateDischarge, false);
+      return pageNames.fileClaimEarly;
+    } else if (differenceBetweenDatesInDays <= 180) {
+      saveDischargeDate(dateDischarge, true);
+      return pageNames.fileBDD;
+    }
   }
   saveDischargeDate();
-  return pageNames.unableToFileBDD;
+  return differenceBetweenDatesInDays < 0 ? null : pageNames.unableToFileBDD;
```

Comment:
Optionally, I think you could flatten this list of conditions a bit by moving the `differenceBetweenDatesInDays < 0` condition to the top w/an early return.

```js
if (differenceBetweenDatesInDays < 0) {
    saveDischargeDate();
    return null;
}

if (differenceBetweenDatesInDays < 90) {
  saveDischargeDate(dateDischarge, false);
  return pageNames.fileClaimEarly;
}

if (differenceBetweenDatesInDays <= 180) {
  saveDischargeDate(dateDischarge, true);
   return pageNames.fileBDD;
}

saveDischargeDate();
return pageNames.unableToFileBDD;

```

Reply:
Yeah, that's a lot more readable. Thanks!